codec-http2/src/main/java/io/netty/handler/codec/http2/UniformStreamByteDistributor.java

```diff
@@ -97,10 +97,13 @@ public boolean distribute(int maxBytes, Writer writer) throws Http2Exception {
         State state = queue.pollFirst();
         do {
             state.enqueued = false;
-            if (state.streamableBytes > 0 && maxBytes == 0) {
-                // Stop at the first state that can't send. Add this state back to the head of
-                // the queue. Note that empty frames at the head of the queue will always be
-                // written.
+            if (!state.writeAllowed) {
```

Comment:
@nmittler  - We could do this, but boolean should be less memory, and this would be the only reason why we need `windowSize`. Would changing the variable name or adding javadocs on the variable be sufficient?